scripts/update_old_checkpoints.py

```diff
@@ -0,0 +1,156 @@
+        yaml.dump(config, new_file)
+    return old_checkpoint_path
+
+
+def update_checkpoint_path(config: config_type, new_checkpoint_path: str | None) -> str | None:
+    if new_checkpoint_path is not None:
+        if "checkpointed_model" in config:
+            old_path = config["checkpointed_model"]["config"]["checkpoint_path"]
+            config["checkpointed_model"]["config"]["checkpoint_path"] = new_checkpoint_path
+            return old_path
+        else:
+            logging.error("'new_checkpoint_path' is set but no 'checkpointed_model' key found in configuration.")
+            exit(1)
+    return None
+
+
+def rename_keys(config: config_type):
+    model_config = config["model_raw" if "model_raw" in config else "model"]["config"]
+    old_norm_keys = ["attention_norm", "ffn_norm", "lm_head_norm"]
+    new_norm_keys = ["attention_norm_config", "ffn_norm_config", "lm_head_norm_config"]
+    for old_key, new_key in zip(old_norm_keys, new_norm_keys):
+        rename_config_key(model_config, old_key, new_key)
+        rename_config_key(model_config[new_key], "variant_key", "norm_type")
+
+
+def rename_config_key(config: config_type, old_key: str, new_key: str):
+    """
+    Rename a single key in the configuration dictionary.
+
+    Args:
+        config (dict): The configuration dictionary.
+        old_key (str): The old key to be renamed.
+        new_key (str): The new key name.
+    """
+    if old_key in config:
+        config[new_key] = config.pop(old_key)
+    else:
+        logging.warning(f"Key '{old_key}' not found in configuration.")
+
+
+def add_new_keys(config: config_type):
+    model_config = config["model_raw" if "model_raw" in config else "model"]["config"]
+    model_config["use_weight_tying"] = False
+    model_config["use_meta_device"] = False
+
+
+def remove_keys(config: config_type):
+    if "evaluation_subscriber" in config and "experiment_id" in config["evaluation_subscriber"]["config"]:
+        del config["evaluation_subscriber"]["config"]["experiment_id"]
+    if "settings" in config and "experiment_id" in config["settings"]:
+        del config["settings"]["experiment_id"]
+    if (
+        "checkpoint_saving" in config
+        and "checkpoint_saving_execution" in config["checkpoint_saving"]["config"]
+        and "experiment_id" in config["checkpoint_saving"]["config"]["checkpoint_saving_execution"]["config"]
+    ):
+        del config["checkpoint_saving"]["config"]["checkpoint_saving_execution"]["config"]["experiment_id"]
+
+
+def update_model_state_dict(old_model_path: str, new_model_path: str):
+    """
+    Update the model state dictionary by loading the old model and saving it to the new path.
+
+    Args:
+        old_model_path (str): Path to the old model file.
+        new_model_path (str): Path to the new model file.
+    """
+    state_dict = torch.load(old_model_path)
+    if "lm_head.weight" in state_dict:
+        state_dict["transformer.lm_head.weight"] = state_dict["lm_head.weight"]
```

Comment:
How would this behave, if we used weight tying? 
Do we store them twice (i.e., embeddings and lm_head) and then internally replace the lm_head with a reference to the embeddings weights?